### ISSUE TITLE
I cannot make my Debian AppVM into a disposable VM

### ISSUE THREAD

**Issue by e6lk7dqzm83p** [QubesOS/qubes-issues#4160]
### Affected Components
Debian VMs

### Steps to reproduce behavior
Unknown, these issues seem to have been present since I installed Qubes

### Expected Behavior
I am able to create a DispVM out of a Debian AppVm

### Actual Behavior
I am not able to create a DispVM out of a Debian AppVm

### Additional Information
I have tried uninstalling and reinstalling the template.
I believe their is a larger issue with Debian App and Template VMs.

**Comment by andrewdavidwong** [comment]
Also, the headers in your issue were malformed Markdown. I don't know why you're manually changing the issue template to invalidate the syntax, but please don't.

**Comment by andrewdavidwong** [comment]
Possibly related: #3877

**Comment by marmarek** [comment]
Please provide exact steps how you try to create DispVM based on Debian.

**Comment by e6lk7dqzm83p** [comment]
1) Created new AppVM via Qubes Manger (I selected "Create New Qube", named the AppVM deb-dvm, and hit ok)
2) In dom0 I typed `qvm-prefs deb-dvm template_for_dispvms true`

I just retried this with a new Debian AppVM with the same result.

I also just tried this via the terminal in dom0:
1)`qvm-create -t debian-9 -l "red" deb-disp3`
2)`qvm-prefs deb-disp3 template_for_dispvms true`

**Comment by e6lk7dqzm83p** [comment]
Sorry, that's correct I was imprecise.  

I cannot create a DVM Template; once (if?) I am able to do that I will let you know if I can launch a DispVM based off of that template.

Thanks!

**Comment by unman** [comment]
@e6lk7dqzm83p - Can you confirm that you have been able to set the `template_for_dispvms` property?
Are you now able to launch disposableVMs using deb-dvm as a DVMTemplate?

**Comment by e6lk7dqzm83p** [comment]
Hi,
I tried creating another fedora DVMTemplate (fedora-dvm-test) and had the same issue, so now it appears I can't create anymore DVMTemplates.  I already have three DVMTemplates (Whonix, Fedora with VPN, Fedora without VPN).  Perhaps there's a limit on the number of DVMTemplates I ran into?

@unman qvm-prefs show the `template_for_dispvms` to be true for both deb-dvm and fedora-dvm-test

Thank you all for the help!

**Comment by marmarek** [comment]
What exactly do you mean by "can't create anymore DVMTemplates"? What is exact effect? Some error message? Something missing that you'd expect to be there (what exactly)?
See https://www.qubes-os.org/doc/dispvm-customization/

**Comment by e6lk7dqzm83p** [comment]
Hi @marmarek sorry for my clumsy descriptions.

When I created the DVMTemplates I mentioned above (and set template_for_dispvm as true) I do not have a way to launch a DispVM through the application launcher.  While all my other DVMTemplates have an entry such as "Disposable: whonix-ws-dvm" or "Disposable: fedora-28-dvm" the recent ones that I've created (deb-dvm and fedora-dvm-test) have no such entry and I'm unable to launch a DispVM based off of those templates.  

When I ran this code in the dom0 terminal:
`qvm-run --dispvm=deb-dvm --service qubes.StartApp+xterm' 

I'd get a notification that the domain dispxxxx is starting, then a notification that domain dispxxxx has started, then a notification that domain dispxxxx has halted.

Does that help explain the issue better?

**Comment by unman** [comment]
@e6lk7dqzm83p Can you check that you can run other applications in the new disposableVM from dom0, and that you can use it as target for open-in-dispvm from other qubes? It's possible that there's a specific issue with xterm.

I hadn't expected that menu items would be created when you set a qube as template_for_dispvm, given that you can use any qube for this purpose. (My aim is to reduce the menu, rather than add to it, so I'd have deleted those items anyway.) It's simple enough to create custom launchers as required. 
I'll check to see if this is specific to Debian or applies to Fedora based qubes also.

**Comment by unman** [comment]
Setting the **template_for_dispvm** property does not seem to create a menu item in Xfce or KDE in either Fedora or Debian for me. Is this expected and desirable behaviour? It is for me. It's simple to create a custom launcher if needed. 
N.B The new DisposableVM **does** appear in the drop down generated for qvm-open etc, so is usable as a target.

Confirmed that for _some_ of my Debian dvms, opening xterm demonstrates the behaviour reported by @e6lk7dqzm83p - other applications open as expected, and they function properly as targets from other Qubes. Further work need to identify reason for this.
This is a distinct and specific problem - I've opened a separate issue

**Comment by e6lk7dqzm83p** [comment]
@unman Firefox worked, running
`qvm-run --dispvm=deb-dvm --service qubes.StartApp+firefox' behaved as expected.

When I changed the default dispvm for an AppVM it did indeed open up a Debian DispVM (it should be noted that launching the terminal for the DispVM was done via the terminal in dom0 as, for some reason, I have an issue getting the Qubes Domain tray icon to launch the terminal in debian AppVMs; I seem to have these weird Qubes/Debian issues).

To answer your question:
> I hadn't expected that menu items would be created when you set a qube as template_for_dispvm, given that you can use any qube for this purpose. (My aim is to reduce the menu, rather than add to it, so I'd have deleted those items anyway.) 

All the other DispVMs that I created show up in the Qubes menu as "Disposable: [DVM Template]" (e.g. "Disposable: whnoix-ws-dvm").

I have noticed some other Debian vs. Fedora AppVM behaviors, so maybe it's all related?

Thank you for the help/response; sorry for being slow I've been slammed at work and just got a chance to update.

**Comment by rustybird** [comment]
> When I ran this code in the dom0 terminal:
> `qvm-run --dispvm=deb-dvm --service qubes.StartApp+xterm'
> 
> I'd get a notification that the domain dispxxxx is starting, then a notification that domain dispxxxx has started, then a notification that domain dispxxxx has halted.

You can debug this by adding `--pass-io` after `qvm-run`. It will probably print "applications/xterm.desktop not found", because the argument to `qubes.StartApp` should be the basename of the .desktop file (not the executable), and Debian-based VMs have their xterm definition in debian-xterm.desktop for some reason. Try `qubes.StartApp+debian-xterm`.

**Comment by unman** [comment]
@e6lk7dqzm83p Are you happy to close this now?
There is a separate bug to capture the inability to start a terminal in Debian based qubes. (#4203)
The lack of a menu item is a separate issue, which is a consequence of the move to diverse DVMTemplates. It affects both Fedora and Debian DVMTemplates.
If you want to raise that as a feature request, please open a new issue or raise it on the qubes-devel mailing list.
If you think that an entry in the docs would be helpful, please feel free to edit https://www.qubes-os.org/doc/dispvm/

**Comment by e6lk7dqzm83p** [comment]
Hi yes, thank you. 

I will post an issue about the DVMTemplates and the menu.

Sorry again for the delay.

**Comment by airelemental** [comment]
To make the app menu entries launch dispXXX as opposed to the DVM template, you have to also run in dom0 `qvm-features deb-dvm appmenus-dispvm 1` and then restart or logout or something (it doesn't update the entries immediately).

**Comment by andrewdavidwong** [comment]
Closing this as "resolved." If you believe the issue is not yet resolved, or if anyone is still affected by this issue, please leave a comment, and we'll be happy to reopen this. Thank you.

**Comment by e6lk7dqzm83p** [comment]
@airelemental thank you for that!

@unman I'm not sure this rises to a feature request as the menu creation is inconsistent: the first two DVMs I created (three total with the Whonix DVM that was created on installation of Qubes) created menu items for the DVMs, after that I just got the DVM template.  I view this inconsistency as a bug.

@andrewdavidwong, I'm happy to keep this marked resolved, I just wanted to add the final points above (sorry I was traveling).  If @unman feels that the inconsistent app menu entry creation warrants keeping this open I don't object to that either.

Thank you all!
